### PR TITLE
sortSearch improvements (fixes #4098)

### LINKED ISSUE #4098: Sort and filter refinements
Some things I noticed.

Motions:
- [ ] "no workflows set" is not possible and should be removed.
- [ ] states should be splitted by first and last states. Last states are all states, with no following states, first states are all others.
- [ ] motion block filter should hide, if no blocks are in the db.
- [ ] if multiple filters are selected, there must be the plural (2 FILTERS)
- [ ] when sorting descending for identifier, the order should be (empty)->4->3->2->1. Currently empty identifier are always on the bottom

Assignments:
- [ ] What is the entry "Assignment" in the sort-menu? Maybe "Title" would be better.

Users:
- [ ] The group-filter is missing

Agenda:
- [ ] The filter-menu is missing

Should be done sometime..

### PR DESCRIPTION
**Pull request by MaximilianKrambach**
- "no workflows set" removed
-  motion block filter hides if there is no motion block
- ￼inserted the group filter
- fixed display of agenda filter menu
- show plural if multiple filters are selected, show filtered amount
- recommendation, done/not done status filter

@FinnStutzenstein 

<S>(The only open thing I could not handle was the `personal notes`, `Favorite` conundrum)</s>

### CODE REVIEW

**Comment by MaximilianKrambach** [comment]
After some communication with @emanuelschuetze I fixed some filters and added (badly layouted) 'clear filter/clear all filters' buttons

File: client/src/app/shared/components/os-sort-filter-bar/os-sort-filter-bar.component.scss [hunk 1]
```diff
@@ -8,3 +8,24 @@
 span.right-with-margin {
     margin-right: 25px;
 }
+
+.flex-spaced {
+    display: flex;
+    justify-content: space-between;
+}
+.filter-count {
```
Comment (tsiegleauq) [inline client/src/app/shared/components/os-sort-filter-bar/os-sort-filter-bar.component.scss]: line spacing

File: client/src/app/shared/components/os-sort-filter-bar/os-sort-filter-bar.component.scss [hunk 2]
```diff
@@ -8,3 +8,24 @@
...
+    text-align: right;
+    margin-right: 10px;
+    margin-left: 10px;
+    min-width: 50px;
+}
+.current-filters {
```
Comment (tsiegleauq) [inline client/src/app/shared/components/os-sort-filter-bar/os-sort-filter-bar.component.scss]: line spacing

File: client/src/app/site/agenda/components/agenda-list/agenda-list.component.html [hunk 3]
```diff
@@ -13,6 +13,7 @@
     </div>
 
 </os-head-bar>
+<mat-drawer-container class="on-transition-fade">
```
Comment (tsiegleauq) [inline client/src/app/site/agenda/components/agenda-list/agenda-list.component.html]: could fail in travis, try to re-align that

File: client/src/app/site/assignments/services/assignment-sort-list.service.ts [hunk 4]
```diff
@@ -10,8 +10,8 @@ export class AssignmentSortListService extends SortListService<ViewAssignment> {
         sortProperty: 'assignment',
         sortAscending: true,
         options: [
-            { property: 'agendaItem', label: 'agenda Item' },
-            { property: 'assignment' },
+            // { property: 'agendaItem', label: 'Agenda item' },
```
Comment (tsiegleauq) [inline client/src/app/site/assignments/services/assignment-sort-list.service.ts]: if not required anymore, remove

File: client/src/app/site/motions/services/motion-filter-list.service.ts [hunk 5]
```diff
@@ -19,55 +20,58 @@ export class MotionFilterListService extends FilterListService<Motion, ViewMotio
...
-        // TODO favorite (attached to user:whoamI!)
-        // TODO personalNote (attached to user:whoamI!)
-    ];
+    public recommendationFilterOptions: OsFilter = {
+        property: 'recommendation',
+        label: 'Recommendation',
+        options: []
+    };
 
     public constructor(
```
Comment (tsiegleauq) [inline client/src/app/site/motions/services/motion-filter-list.service.ts]: doc

File: client/src/app/site/motions/services/motion-filter-list.service.ts [hunk 6]
```diff
@@ -77,6 +81,16 @@ export class MotionFilterListService extends FilterListService<Motion, ViewMotio
         this.subscribeComments();
     }
 
+    // does not work, some cloning error. I want to:
```
Comment (tsiegleauq) [inline client/src/app/site/motions/services/motion-filter-list.service.ts]: add todo

File: client/src/app/site/agenda/components/agenda-list/agenda-list.component.html (hunk 3 above)
Comment (MaximilianKrambach) [reply client/src/app/site/agenda/components/agenda-list/agenda-list.component.html]: so apparently prettify-write does not all of these things

**Comment by MaximilianKrambach** [comment]
The missing 'Favorite'/'Personal notes' filters will be handled in #4120

**Pull request merged**